### ISSUE TITLE
Add Unitary Group

### ISSUE THREAD

**Issue by HiperMaximus** [QUVA-Lab/escnn#90]
I think adding the unitary group U(n) and special unitary SU(n) would be interesting as they are 'bigger' than O(n) and SO(n) (and also connected which might be interesting), and many physical models are invariant under the unitary group. I'm uncertain if it would necessitate adding support for complex numbers.

**Comment by maxxxzdn** [comment]
I think at this point, _escnn_ is focused on all subgroups of the Euclidean group E(n), so adding support for unitary groups might be a heavy lift. Perhaps one can develop _uscnn_ for that matter :)

**Comment by Gabri95** [comment]
Indeed, we currently don't support the U(n) and SU(n) groups but that is not too hard and doens't require complex numbers.
As we argue in our paper, using complex representations is either unnecessary or can lead to underparameterized models since computer architectures operate with real numbers.

In this sense, you can think of U(n) as a subgroup of O(2n). As far as I know, the main challenges here are:
1- converting the U(n) complex representations of interest into real ones. For U(n), a simple "realification" should work out of the box (convert each complex number to a 2x2 real matrix)
2-  find an application requiring U(n) or SU(n) equivariance.

If someone is interested in implementing this kind of architeture I can provide some guidance

Best,
Gabriele
